lexicon/modification.py

```diff
@@ -0,0 +1,156 @@
+
+
+class StaticLexiconJob(Job):
+    """
+    Create a bliss lexicon file from a static Lexicon.
+
+    Supports optional sorting of phonemes and lemmas.
+
+    Example for a static lexicon:
+
+    .. code: python
+
+        static_lexicon = lexicon.Lexicon()
+        static_lexicon.add_lemma(
+            static_lexiconicon.Lemma(
+                orth=["[SILENCE]", ""],
+                phon=["[SILENCE]"],
+                synt=[""],
+                special="silence",
+                eval=[""],
+            )
+        )
+        static_lexicon.add_lemma(
+            static_lexiconicon.Lemma(
+                orth=["[UNKNOWN]"],
+                phon=["[UNKNOWN]"],
+                synt=[["<UNK>"]],
+                special="unknown",
+            )
+        )
+        static_lexicon.add_phoneme("[SILENCE]", variation="none")
+        static_lexicon.add_phoneme("[UNKNOWN]", variation="none")
+    """
+
+    def __init__(self, static_lexicon, sort_phonemes, sort_lemmas, compressed=False):
+        """
+        :param lexicon.Lexicon static_lexicon: A Lexicon object
+        :param bool sort_phonemes: sort phoneme inventory alphabetically
+        :param bool sort_lemmas: sort lemmas alphabetically based on first orth entry
+        :param bool compressed: compress final lexicon
+        """
+        self.static_lexicon = static_lexicon
+        self.sort_phonemes = sort_phonemes
+        self.sort_lemmas = sort_lemmas
+
+        self.out_bliss_lexicon = self.output_path(
+            "lexicon.xml.gz" if compressed else "lexicon.xml"
+        )
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        lex = lexicon.Lexicon()
+        if self.sort_phonemes:
+            sorted_phoneme_list = [
+                (k, self.static_lexicon.phonemes[k])
+                for k in sorted(self.static_lexicon.phonemes.keys())
+            ]
+            for phoneme_tuple in sorted_phoneme_list:
+                lex.add_phoneme(symbol=phoneme_tuple[0], variation=phoneme_tuple[1])
+        else:
+            lex.phonemes = self.static_lexicon.phonemes
+
+        if self.sort_lemmas:
+            lemma_dict = {}
+            for lemma in self.static_lexicon.lemmas:
+                # sort by first orth entry
+                lemma_dict[lemma.orth[0]] = lemma
+            lex.lemmas = [lemma_dict[key] for key in sorted(lemma_dict.keys())]
+        else:
+            lex.lemmas = self.static_lexicon.lemmas
+
+        write_xml(self.out_bliss_lexicon.get_path(), lex.to_xml())
+
+
+class MergeLexiconJob(Job):
+    """
+    Merge multiple bliss lexica into a single bliss lexicon.
+
+    Phonemes and lemmas can be individually sorted alphabetically or kept as is.
+
+    When merging a lexicon with a static lexicon, putting the static lexicon first
+    and only sorting the phonemes will result in the "typical" lexicon structure.
+    """
+
+    def __init__(self, bliss_lexica, sort_phonemes, sort_lemmas, compressed=False):
```

Comment:
the ` sort_phonemes` and  `sort_lemmata` params can have a default value and it should imho be `False`

Maybe we want to add a warning that if sorting of lemmata is enabled, any alignment based on the old lexicon might be invalidated

Reply:
only sorting of phonemes might be affected. sorting lemmas will not break alignments.

Reply:
no, in fact the other way round. changing the order of the lemmata might in some cases change the order of the allophone table. (e.g. when no complete allophone file is used)